packages/flutter/lib/src/material/popup_menu.dart

```diff
@@ -145,17 +207,111 @@ class _PopupMenuRoute<T> extends PopupRoute<T> {
   Color get barrierColor => null;
 
   Widget buildPage(BuildContext context, Animation<double> animation, Animation<double> forwardAnimation) {
-    return new _PopupMenu(route: this);
+    int selectedIndex = -1;
+    if (initialValue != null) {
+      for (int i = 0; i < items.length; i++)
+        if (initialValue == items[i].value) {
+          selectedIndex = i;
+          break;
+        }
+    }
+    final Size screenSize = MediaQuery.of(context).size;
+    return new SizedBox(
+      width: screenSize.width,
+      height: screenSize.height,
+      child: new CustomOneChildLayout(
+        delegate: new _PopupMenuRouteLayout(position, selectedIndex),
+        child: new _PopupMenu(route: this)
+      )
+    );
   }
 }
 
-Future showMenu({ BuildContext context, ModalPosition position, List<PopupMenuItem> items, int elevation: 8 }) {
+/// Show a popup menu that contains the [items] at [position]. If [initialValue]
+/// is specified then the first item with a matching value will be highlighted
+/// and the value of [position] implies where the left, center point of the
+/// highlighted item should appear. If [initialValue] is not specified then position
+/// implies the menu's origin.
+Future showMenu({
+  BuildContext context,
+  ModalPosition position,
+  List<PopupMenuItem> items,
+  dynamic initialValue,
+  int elevation: 8
+}) {
+  assert(context != null);
+  assert(items != null && items.length > 0);
   Completer completer = new Completer();
   Navigator.push(context, new _PopupMenuRoute(
     completer: completer,
     position: position,
     items: items,
+    initialValue: initialValue,
     elevation: elevation
   ));
   return completer.future;
 }
+
+/// A callback that is passed the value of the PopupMenuItem that caused
+/// its menu to be dismissed. Null if the menu was dismissed without
+/// selecting an item.
+typedef void PopupMenuDismissed(dynamic value);
```

Comment:
Here's my rationale for 'onDismissed'
- Whether or not an item was selected, what triggers this callback is the menu being dismissed
- Providing an onSelected callback would be useful, but sometimes it's useful to know that the user dismissed the menu but didn't select an item ("do you accept the license: yes/no").
- valueChanged<T> doesn't match the callback very well since dismissing the menu without making a selection, or selecting the initialValue item, doesn't correspond to the a value change.